### PR TITLE
fix: roll review notes back to the persisted list

### PR DESCRIPTION
**Pull request by AmethystLiang**
## ELI5

If saving a review note to disk fails twice in a row, the app used to keep showing a note that was never actually saved. Now it always falls back to what is really on disk. Plus, the note schema now rejects junk (blank body, empty ids) and stops rejecting valid file-level notes from mobile.

## What Changed

Follow-up to #14112, addressing CodeRabbit review feedback left on that PR after it merged.

**1. Roll review notes back to the persisted list** (`src/renderer/src/store/slices/diffComments.ts`)

`rollback` restored `mutation.previous`, which is itself an optimistic list when writes queue up. With two queued writes that both fail, the second rollback restored the first mutation's optimistic list while storage still held the list from before both mutations, so the renderer showed notes that were not durable.

- Added `durableCommentsByQueue`: the last list actually written per persist queue. Seeded from the pre-mutation list when the queue is idle (nothing in flight means state matches disk), updated after each successful write, dropped when the queue drains.
- `enqueuePersist` now resolves a `PersistOutcome` instead of rejecting, so a failed write carries the durable list back to its own caller without breaking the chain.
- New `commitMutation` is the single commit path (queue the write, roll back to the durable list on failure), replacing seven duplicated try/catch blocks. The existing identity guard in `rollback` is unchanged, so a newer mutation still wins.

**2. Enforce the `DiffComment` invariants** (`src/shared/diff-comment-schema.ts`)

`id` / `worktreeId` / `filePath` require content, `body` is trimmed and must be non-empty, `sentAt` must be positive (matching the renderer, which already treats non-positive as unsent), and `startLine <= lineNumber` is refined. `lineNumber` moved from `.positive()` to `.int().min(0)`: mobile uses `lineNumber: 0` for file-level notes, so the host was rejecting valid ones.

**3. Validate persisted note elements** (`src/shared/folder-workspaces.ts`)

`normalizeFolderWorkspaces` accepted any array via `Array.isArray`, rehydrating malformed notes straight past the new schema boundary. It now parses each element with the shared schema (`parsePersistedDiffComments`) and drops invalid ones.

## Why

Review notes are user-authored content that is easy to lose silently: an optimistic list that looks saved but is not means the user closes the diff believing their notes are on disk. Tracking the durable list per queue is the smallest change that keeps rollback honest without an extra read from storage.

The schema fixes close the other half: the boundary now rejects what the renderer would never produce, and load-time normalization drops legacy junk so it is never re-sent to the host.

## Linked Issue

Follow-up to #14112 (STA-3958). No separate issue: this addresses review feedback on that PR.

## Visual Proof

`N/A` — no UI change. The only user-visible difference is in a failure path (notes now revert to the persisted list rather than showing an unsaved note), which is covered by the new store tests.

## Testing

- `vitest` over the diff-comment, folder-workspace, persistence, and selector suites: 512 passed.
- Verified the new "two queued writes fail" test is a real regression test: temporarily reverting `outcome.durable` to `mutation.previous` makes it fail, then restored.
- `tsc --noEmit` on the node, web, and cli projects: clean.
- `oxlint` on all touched files: clean. `check:max-lines-ratchet`: OK (no new suppressions).

New tests:
- `folder-workspace-diff-comments.test.ts` — two queued writes both fail (rolls back to the empty persisted list); first write succeeds and second fails (keeps the persisted note instead of reverting).
- `src/shared/diff-comment-schema.test.ts` — boundary cases for empty fields, whitespace body, `startLine` range, `lineNumber: 0`, non-positive `sentAt`.
- `src/shared/folder-workspaces.test.ts` — mixed valid/invalid persisted `diffComments`.

- [x] I manually tested these changes locally
- [x] Automated tests added/updated, or explained why not below

## Review

- **Backwards compatibility / remote wire**: the schema tightening is host-side validation on `folderWorkspace.update` (IPC and RPC), so it reaches older paired clients. Notes with an empty body/id or non-positive `sentAt` are now rejected rather than stored. The UI cannot produce those (the popover gates submit on non-empty text; the store normalizes `sentAt`/`startLine` before persisting), and change 3 drops such legacy records at load so they are not re-sent. The `lineNumber: 0` change is a loosening and is strictly compatible. Flagging in case a reviewer prefers coercion over rejection here.
- **Mobile**: `lineNumber: 0` file-level notes are now accepted by the desktop host instead of rejected.
- **Cross-platform / SSH**: no platform- or path-dependent code; the folder-workspace and worktree paths both route through the same queue as before, local and RPC alike.
- **Performance**: one extra `Map` entry per in-flight persist queue, cleared when the queue drains.

## AI Disclosure

Claude Opus 5 (Claude Code).

Made with [Orca](https://github.com/stablyai/orca) 🐋

### CODE REVIEW

File: src/renderer/src/store/slices/diffComments.ts [hunk 1]
```diff
@@ -154,36 +154,55 @@ function settingsForWorktreeOwner(state: AppState, worktreeId: string): AppState
...
+            state.worktreesByRepo[getRepoIdFromWorktreeId(worktreeId)]?.find(
+              (w) => w.id === worktreeId
+            )?.diffComments ?? []
+          ).map(normalizeDiffComment)
+    try {
+      await (scope?.type === 'folder'
+        ? persist(state, state.settings, worktreeId, latest, folderExecutionHostId)
+        : persist(state, settingsForWorktreeOwner(state, worktreeId), worktreeId, latest))
+    } catch (error) {
+      return { ok: false, error, durable: durableCommentsByQueue.get(queueKey) }
```
[the remark this answers is not in the document]
Comment (AmethystLiang) [reply src/renderer/src/store/slices/diffComments.ts]: Fixing. Will be in the next commit
